sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -619,3 +620,35 @@ object DataSourceStrategy {
     (nonconvertiblePredicates ++ unhandledPredicates, pushedFilters, handledFilters)
   }
 }
+
+
+/**
+ * Support for recalculating table statistics if table statistics are not available.
+ */
+class DetermineTableStats(session: SparkSession) extends Rule[LogicalPlan] {
+
+  private val sessionConf = session.sessionState.conf
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+    // For the data source table, we only recalculate the table statistics when it creates
+    // the CatalogFileIndex using defaultSizeInBytes. See SPARK-25474 for more details.
```

Comment:
`when it creates the CatalogFileIndex using defaultSizeInBytes` -> `when the table stats are not available`

Reply:
Done